tests/end2end/test_session.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2023 IBM Corp. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+End2end tests for the 'zhmc session' command group.
+"""
+
+from __future__ import absolute_import, print_function
+
+import os
+import re
+from requests.packages import urllib3
+
+# pylint: disable=line-too-long,unused-import
+from zhmcclient.testutils import hmc_definition  # noqa: F401, E501
+# pylint: enable=line-too-long,unused-import
+
+from .utils import zhmc_session_args, run_zhmc
+
+urllib3.disable_warnings()
+
+
+def test_session_success(hmc_definition):  # noqa: F811
+    # pylint: disable=redefined-outer-name
+    """
+    Test successful creation and deletion of an HMC session.
+    """
+
+    # Create the session
+
+    # Set to True to debug 'zhmc session create'
+    pdb = False
+
+    zhmc_args = zhmc_session_args(hmc_definition) + ['session', 'create']
+
+    # The code to be tested
+    rc, stdout, stderr = run_zhmc(zhmc_args, pdb=pdb)
+
+    if not pdb:
```

Comment:
style: why check something you know to be false? This is somehow confusing to the reader.
If you want to have the ability to change `pdb` manually sometimes, then I would suggest you refactor your code, and have a helper method where you pass `pdb` as argument.